elements/pl-checkbox/pl-checkbox.py

```diff
@@ -30,6 +30,11 @@ def prepare(element_html, data):
     pl.check_attribs(element, required_attribs, optional_attribs)
     name = pl.get_string_attrib(element, 'answers-name')
 
+    if name in data['answers_names']:
+        raise Exception('duplicate answer-name attribute: %s' % name)
+    else:
+        data['answers_names'][name] = True
```

Comment:
Instead of a dictionary, I wonder if it may be better to store this as a list instead, since the `True` value brings no meaning at all. Alternatively, if there is any useful information here (e.g., something that can add information to the exception message to help the instructor find the duplicate), the `True` value could be replaced with that, but I don't think this would make a big difference.

Reply:
I think storing as a list would be a mistake, since the data structure would allow for insertion of duplicates. Making this a dict, although a little weird, makes it clear that keys should be unique.

Reply:
Actually, I'm kinda flip-flopping on this. Since this logic is now handled by a helper function, using a list might not make a difference (since users won't really see this). Do you have any thoughts @nwalters512 ?

Reply:
I think a dictionary still makes sense here. What we *really* want is a set, but a dictionary with `True` values is as close as we can get with JSON.